packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -69,47 +67,43 @@ export default function TemplateSwitcher( {
 
 	const { currentTheme, templates, templateParts } = useSelect(
 		( select ) => {
-			const { getCurrentTheme, getEntityRecord } = select( 'core' );
+			const { getCurrentTheme, getEntityRecords } = select( 'core' );
+
+			const homeId = getEntityRecords( 'postType', 'wp_template', {
+				resolved: true,
+				slug: 'front-page',
```

Comment:
`front-page` is not necessarily the home template. You need to make the same API call `navigate-to-link` makes.

Reply:
Done in ff313c4. Tho in the long run, I think we should modify the `wp_template` endpoint, or altenatively write a new endpoint that returns the template for a given entity or query.